rfd/0173-terraform-machine-id.md

```diff
@@ -0,0 +1,326 @@
+> Actions/GitLab CI) or via a dedicated service (Terraform Cloud/Spacelift).
+
+The user is expected to know the Teleport basics and be able to perform advanced setup actions such as
+creating roles and bot resources for MachineID. Some setup complexity is not an issue for this persona.
+
+#### The "getting started" users
+
+We have two personas:
+
+- The prospect:
+  > I am not yet a Teleport user and want to test what the Teleport IaC experience looks like before using/buying Teleport.
+  > IaC will be mandatory for me because I am subject to various internal policies and I must check if Teleport is
+  > compatible with my IaC policies.
+  > I have prior Terraform experience.
+
+- The existing manual user
+  > I am an existing Teleport user that have been managing Teleport resources manually.
+  > Today I am investigating how to make my setup more robust by leveraging an IaC tool.
+  > I might not have prior Terraform experience.
+
+Both personas don't have (yet) dedicated infrastructure such as CI runners to deploy the Terraform code.
+They want to prototype and validate everything works using the local computer.
+
+Both personas have very little knowledge about Teleport and don't know what MachineID is and how to configure it.
+We want those users to succeed as fast as possible without having to read all the MachineID docs. When they will go
+to production, they will hopefully turn into [existing advanced users](#the-existing-advanced-user) and setup CI/CD
+pipelines for their deployment.
+
+### Implementation
+
+We will introduce two new mechanisms:
+- MachineID joining in the Terraform provider
+- Terraform Resource bootstrapping in `tctl`
+
+#### MachineID joining
+
+The Terraform provider will be able to natively join a Teleport cluster using MachineID when provided with a token and join method.
+The users will not need to install or configure `tbot`.
+
+The Terraform provider will run an in-process `tbot` instance and generate a client with automatic cert-renewal.
+The `tbot` instance will not persist its certificates on the disk.
+
+A typical GitHub actions provider configuration would look like:
+
+```hcl
+provider "teleport" {
+  addr               = "mytenant.teleport.sh:443"
+  joining = {
+    token  = "gha-runner"
+    method = "github"
+  }
+}
+```
+
+As Teleport does not support yet reusable bot tokens so the `token` join method will not be supported.
+_Note: this might change in the future with the `BotInstance` resource decoupling the bot from the
+secret token._
+
+Implementation-wise, this would reuse the
+same [in-process tbot library](https://github.com/gravitational/teleport/tree/rfd/173-terraform-machine-id/integrations/operator/embeddedtbot)
+we used for the Teleport operator.
+
+##### Note regarding CGO
+
+The Terraform provider is currently built without CGO but `tbot` depends on `lib/auth`, which depends on `lib/services`
```

Comment:
I don't think this note is relevant anymore taking Tim's comment below into account?